src/frontend/screens/DeviceNaming/index.tsx

```diff
@@ -0,0 +1,148 @@
+import * as React from 'react';
+import {NativeStackScreenProps} from '@react-navigation/native-stack';
+import {DeviceNamingList} from '../../Navigation/ScreenGroups/DeviceNamingScreens';
+import CoMapeoText from '../../images/CoMapeoText.svg';
+import {
+  StyleSheet,
+  View,
+  TextInput,
+  KeyboardAvoidingView,
+  Keyboard,
+} from 'react-native';
+import {BLACK, LIGHT_GREY, MEDIUM_GREY, RED} from '../../lib/styles';
+import {Text} from '../../sharedComponents/Text';
+import {Button} from '../../sharedComponents/Button';
+import {TouchableWithoutFeedback} from 'react-native-gesture-handler';
+import {defineMessages, useIntl} from 'react-intl';
+
+const m = defineMessages({
+  header: {
+    id: 'screens.DeviceNaming.header',
+    defaultMessage: 'Add a name for your device',
+  },
+  description: {
+    id: 'screens.DeviceNaming.description',
+    defaultMessage:
+      "Guidance on how to choose a name and what it's for. You can edit this later",
+  },
+  addName: {
+    id: 'screens.DeviceNaming.addName',
+    defaultMessage: 'Add Name',
+  },
+});
+
+export const DeviceNaming = ({
+  navigation,
+}: NativeStackScreenProps<DeviceNamingList, 'DeviceNaming'>) => {
+  const [name, setName] = React.useState('');
+  const [errorTimeout, setErrorTimeout] = useErrorTimeout();
+  const invalidName = name.length === 0 || name.length > 60;
+  const {formatMessage: t} = useIntl();
+
+  function setNameWithValidation(nameValue: string) {
+    if (nameValue.length > 60) {
+      setErrorTimeout();
+      return;
+    }
+    setName(nameValue);
+  }
+
+  function handleAddNamePress() {
+    if (invalidName) {
+      setErrorTimeout();
+      return;
+    }
+
+    navigation.navigate('Success', {deviceName: name});
+  }
```

Comment:
mentioned in other comment, but i think the persistence of the device name should happen here and you should only navigate to the success screen after that happens without error

Reply:
There is a semi annoying complication with conditional rendering of screens. If we set the name here the name is set, and these screen will not be rendered anymore (including the next screen). The conditional rendering is nice because it is declarative, aka the screens are based of the state of the device Name. 

We could make it imperative, and set the name here, and just navigate to the success screen, and then navigate to the map screen. It would just require the app to imperatively check the state on every open of the app. Aka, on the open of the app we would have to do the following at the top level:

```ts
// imperative
const deviceId = getDeviceInfo()
if (!deviceId) {
navigate("deviceNaming)
} else {
navigate("map)
}
```

Reply:
Also ff we do it imperatively, after the hit `go to map`, they are navigated to the map. But if they hit the back button, those screen will be on the nav stack, so they would be able to go back to those screen. We could do a check on those screens:
``` ts
// in all device naming screens
useLayoutEffect(()=>{
   if(!!deviceId) navigate("map")
})
```

that feels equally as clunky as the AppState Event listener